src/bindgen/ir/function.rs

```diff
@@ -23,17 +23,20 @@ use crate::bindgen::writer::{Source, SourceWriter};
 #[derive(Debug, Clone)]
 pub struct Function {
     pub path: Path,
+    pub self_type_path: Option<Path>,
```

Comment:
Please add a comment saying something like "If we're a method, this will be `Some`, and the path will be the path of the type in the `impl` block" or such.

It'd be nice to make something more similar to what associated constants do, but it may be overkill for this.